src/main/java/org/alfresco/repo/security/authentication/identityservice/IdentityServiceDeploymentFactoryBean.java

```diff
@@ -51,7 +55,14 @@ public void setIdentityServiceConfig(IdentityServiceConfig config)
     public KeycloakDeployment getObject() throws Exception
     {
         KeycloakDeployment deployment = KeycloakDeploymentBuilder.build(this.identityServiceConfig);
-        
+
+        // Set client with custom timeout values. This can be removed if the future versions of Keycloak accept timeout values through the config.
+        HttpClient client = new HttpClientBuilder()
+                .establishConnectionTimeout(2, TimeUnit.SECONDS)
+                .socketTimeout(2, TimeUnit.SECONDS)

```

Comment:
1. Are there other timeouts as well? What if we could establish the connection real quick and the TCP connection is stable, but the other end simply does not respond? There should be another timeout on the HTTP level
2. Not sure if there are situations where we need to increase this timeout. perhaps we should read these values from prop files in the end. Just to have a possibility for support to increase them in some edge case scenarios